### PR TITLE
chore: replace repository name and owner with Github context variables

### PR DESCRIPTION
**Pull request by matthiashermsen**

### CODE REVIEW

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -76,7 +76,7 @@ jobs:
       if: ${{ steps.get_next_version.outputs.hasNextVersion == 'true' }}
       uses: docker/build-push-action@v5
       env:
-          DOCKER_REPOSITORY: ghcr.io/thenativeweb/get-next-version
+          DOCKER_REPOSITORY: ghcr.io/${{ github.repository }}
```
Comment (goloroden) [inline .github/workflows/release.yml:79]: Why aren't you using the `repository_owner` and the `event.repository.name` here as well (or, vice-versa, why aren't you using this field above?

Shouldn't both places be the same value?

**Review by goloroden — CHANGES_REQUESTED** [review]
Hi @matthiashermsen 👋 

Thanks for the PR.

I've taken a look at it, but TBH I'm not convinced yet. In the end, it makes it harder to read which repository gets targeted, since now you don't have the single possible value hard-coded, but need to defer it from three different variables.

Am I missing a benefit here?

**Comment by matthiashermsen** [comment]
> Hi @matthiashermsen 👋
> 
> Thanks for the PR.
> 
> I've taken a look at it, but TBH I'm not convinced yet. In the end, it makes it harder to read which repository gets targeted, since now you don't have the single possible value hard-coded, but need to defer it from three different variables.
> 
> Am I missing a benefit here?

Hi @goloroden ,
IMHO this makes it easier to reuse workflows. If one changes repository names / ownerships the workflows still work because Github will inject the correct values.

File: .github/workflows/release.yml (hunk 1 above)
Comment (matthiashermsen) [reply .github/workflows/release.yml:79]: There is no special reason, I just replaced the "hard-coded" parts with Github variables.

- `github.event.repository.name` returns the repository name ( without owner )
- `github.repository_owner` returns the repository owner
- `github.repository` returns both with a slash

So you could use 

`${{ github.repository_owner }}/${{ github.event.repository.name }}`

too, yeah :)

**Comment by goloroden** [comment]
Sorry for getting back to you that late … I have thought about this one, and TBH I don't see the benefit here, especially compared to that it makes it harder to guess what values are actually being used.

This makes debugging more work, for something that happens once in a lifetime.

Hence I'm going to close this one (without merging this).

I hope you aren't mad at me for this.

**Comment by matthiashermsen** [comment]
> Sorry for getting back to you that late … I have thought about this one, and TBH I don't see the benefit here, especially compared to that it makes it harder to guess what values are actually being used.
> 
> This makes debugging more work, for something that happens once in a lifetime.
> 
> Hence I'm going to close this one (without merging this).
> 
> I hope you aren't mad at me for this.

Sure, no problem :)

**Pull request closed without merging**